cmd/crates/stellar-rpc-client/src/lib.rs

```diff
@@ -799,7 +799,7 @@ impl Client {
 
     ///
     /// # Errors
-    pub async fn get_transaction(&self, tx_id: &Hash) -> Result<GetTransactionResponse, Error> {
+    pub async fn get_transaction(&self, tx_id: &Hash) -> Result<GetTransactionResponseRaw, Error> {
```

Comment:
I think the issue here is that this should still be returning the GetTransactionResponse, but it should be converting from the raw to the non raw here. The raw isn't intended to be used externally I thought.

Reply:
Looking at the code prior to its last change it does appear that the function used to return the raw version, which really begs the question why the two versions exist if we expose the raw version to people anyway: https://github.com/stellar/soroban-rpc/commit/587a229541043ca3b2e3e4514296d6c87625b25c#diff-ef42a2e4c0c5ce57fe0118fafc5db591609074ffcb19fc3a7a8576b32026d423L860-L863

Reply:
I think we should fix this by having all the public fns return the same type, `GetTransactionResponse`, and make GetTransactioResponseRaw unexported. cc @2opremio wdyt?

Reply:
yeah that's what the JS SDK does, e.g. for `(Raw)?LedgerEntryChange` [here](https://github.com/stellar/js-stellar-sdk/blob/master/src/rpc/api.ts#L171-L186).